client/llb/source.go

```diff
@@ -423,13 +429,33 @@ func SharedKeyHint(h string) LocalOption {
 	})
 }
 
+func Differ(t DiffType, required bool) LocalOption {
+	return localOptionFunc(func(li *LocalInfo) {
+		li.Differ = DifferInfo{
+			Type:     t,
+			Required: required,
+		}
+	})
+}
+
+type DiffType string
+
+const DiffNone DiffType = pb.AttrLocalDifferNone
```

Comment:
Can we get some description comments here, especially `DiffNone` as a name feels ambiguous to me.  Maybe something like:
```
const (
  // DiffNone will do no file comparisons, all files in the Local source will
  // be retransmitted.
  DiffNone DiffType = pb.AttrLocalDifferNone

  // DiffMetadata will compare file metadata (size, modified time, mode, owner,
  // group, device and link name) to determine if the files in the Local source need
  // to be retransmitted.  This is the default behavior.
  DiffMetadata DiffType = pb.AttrLocalDifferMetadata
)
```